### PR TITLE
tls options added to net_plugin

### PR DESCRIPTION
**Pull request by dimas1185**
## Change Description

Added 3 new command line arguments to net_plugin to facilitate TLS connection

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [ ] Other

- [x] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [x] API Changes

Added the following configuration options:
--p2p-tls-ca-certificate-file: "Certificate Authority's certificate file used for verifying peers TLS connection when security groups feature enabled"
--p2p-tls-own-certificate-file: "Certificate file that will be used to authenticate running node if TLS is enabled"
--p2p-tls-private-key-file: "Private key file that is used in conjunction with tsl-own-certificate-file for server authorization in TLS connection. Together tls-private-key-file + tsl-own-certificate-file automatically enables TLS-only connection for peers."

## Documentation Additions
- [] Documentation Additions

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -3653,7 +3653,9 @@ namespace eosio {
            "   _lip   \tlocal IP address connected to peer\n\n"
            "   _lport \tlocal port number connected to peer\n\n")
          ( "p2p-keepalive-interval-ms", bpo::value<int>()->default_value(def_keepalive_interval), "peer heartbeat keepalive message interval in milliseconds")
-
+         ( "tls-ca-certificate-file", bpo::value<string>(), "Certificate Authority's certificate file used for verifying peers TLS connection when security groups feature enabled" )
+         ( "tls-own-certificate-file", bpo::value<string>(), "Certificate file that will be used to authenticate running node if TLS is enabled")
+         ( "tls-private-key-file", bpo::value<string>(), "Private key file that is used in conjunction with tsl-own-certificate-file for server authorization in TLS connection. Together tls-private-key-file + tsl-own-certificate-file automatically enables TLS-only connection for peers.")
```
Comment (heifner) [inline plugins/net_plugin/net_plugin.cpp:3658]: I think these should be prefixed with `p2p`

File: tests/generate-certificates.sh [hunk 2]
```diff
@@ -54,17 +54,11 @@ GROUP_SIZE=4
 #overrides default is set
 parse-args "${@}"
 
-echo "*************************************************"
-echo "         generating dh param                     "
-echo "*************************************************"
-#using low values like 128 here and below as this is for unit tests and our goal to save running time. For real applications 2048 recommended
-openssl dhparam -out dh.pem 128
-
```
Comment (dimas1185) [inline tests/generate-certificates.sh:62]: explicit dh params key exchange is less performant than elliptic curve. Google Chrome removed this type of key exchange in 2016 so I don't think we should support it.

File: tests/generate-certificates.sh [hunk 3]
```diff
@@ -54,17 +54,11 @@ GROUP_SIZE=4
...
 echo "*************************************************"
 echo "         generating CA_cert.pem                  "
 echo "*************************************************"
 
-openssl req -newkey rsa:512 -nodes -keyout CA_key.pem -x509 -days ${DAYS} -out CA_cert.pem -subj "/C=US/ST=VA/L=Blocksburg/O=${CA_ORG}/CN=${CA_CN}"
+openssl req -newkey rsa:2048 -nodes -keyout CA_key.pem -x509 -days ${DAYS} -out CA_cert.pem -subj "/C=US/ST=VA/L=Blocksburg/O=${CA_ORG}/CN=${CA_CN}"
```
Comment (dimas1185) [inline tests/generate-certificates.sh:61]: consideration is that real size keys will show real performance overhead

File: plugins/net_plugin/net_plugin.cpp [hunk 4]
```diff
@@ -3653,7 +3653,9 @@ namespace eosio {
            "   _lip   \tlocal IP address connected to peer\n\n"
            "   _lport \tlocal port number connected to peer\n\n")
          ( "p2p-keepalive-interval-ms", bpo::value<int>()->default_value(def_keepalive_interval), "peer heartbeat keepalive message interval in milliseconds")
-
+         ( "p2p-tls-ca-certificate-file", bpo::value<string>(), "Certificate Authority's certificate file used for verifying peers TLS connection when security groups feature enabled" )
+         ( "p2p-tls-own-certificate-file", bpo::value<string>(), "Certificate file that will be used to authenticate running node if TLS is enabled")
+         ( "p2p-tls-private-key-file", bpo::value<string>(), "Private key file that is used in conjunction with tsl-own-certificate-file for server authorization in TLS connection. Together tls-private-key-file + tsl-own-certificate-file automatically enables TLS-only connection for peers.")
```
Comment (brianjohnson5972) [inline plugins/net_plugin/net_plugin.cpp:3658]: update the string references of paramaters with "p2p-" prefix that was added

**Review by brianjohnson5972 — CHANGES_REQUESTED** [review]

**Review by brianjohnson5972 — APPROVED** [review]

**Pull request merged**